drivers/scheduler/k8s/k8s.go

```diff
@@ -2848,10 +2848,22 @@ func (k *K8s) ScaleApplication(ctx *scheduler.Context, scaleFactorMap map[string
 			}
 			newScaleFactor := scaleFactorMap[obj.Name+DeploymentSuffix]
 			*dep.Spec.Replicas = newScaleFactor
-			if _, err := k8sOps.UpdateDeployment(dep); err != nil {
+			var updateError error
+			retries := 5
+			for i := 0; i < retries; i++ {
+				_, updateError := k8sOps.UpdateDeployment(dep)
+				if updateError == nil {
+					break
+				}
+				if i < retries - 1 {
+					logrus.Warnf("Failed to update deployment. Error: %v. Going to retry", updateError)
+					time.Sleep(time.Second * 10)
+				}
+			}
+			if updateError != nil {
```

Comment:
use task.DoRetryWithTimeout function to do retries.

Reply:
thanks for the suggestion, will change